lib/s3storage.php

```diff
@@ -133,7 +134,12 @@ public function writeObject($urn, $stream) {
 		try {
 			$uploader->upload();
 		} catch (AwsException $e) {
-			throw new ObjectStoreWriteException($e->getMessage(), $e->getCode(), $e);
+			/**
+			 * If the error is from AwsException then just wrap the aws error message
+			 * to our exception.
+			 */
+			\OC::$server->getLogger()->logException($e);
```

Comment:
Is it necessary to log the exception here? the message is in the ObjectStoreWriteException and could be logged on the caller side as well.

In terms of proper encapsulation bringing the logger dependency into this class is wrong.
Please explain the need for this. THX

Reply:
The only reason I brought the logger to this class is because of https://github.com/owncloud/files_primary_s3/pull/287#issuecomment-566022732. Yes you are right the caller side can log this as well.

Yes bringing dependency of logger is again something which worried me. I will remove the logging from this pr.

Reply:
Removed the logging from this PR. @DeepDiver1975 Let me know if this looks ok now.